pom.xml

```diff
@@ -612,6 +612,11 @@
                     <groupId>org.jacoco</groupId>
                     <artifactId>jacoco-maven-plugin</artifactId>
                     <version>0.8.11</version>
+                    <configuration>
+                        <excludes>
+                            <exclude>**/target/generated-sources/**</exclude>
+                        </excludes>
```

Comment:
Not sure, if this will work. According to the docs, we can exclude classes but not source paths... (https://www.jacoco.org/jacoco/trunk/doc/report-mojo.html#excludes)

But I found another potential solution (thanks to https://www.baeldung.com/jacoco-report-exclude ).

Apparently, jacoco added a feature in version 0.8.2 (https://github.com/jacoco/jacoco/issues/696) which automatically excludes any classes that are annotated with a annotation named "Generated". We could use something like [javax.annotation.processing.Generated](https://docs.oracle.com/en/java/javase/21/docs/api/java.compiler/javax/annotation/processing/Generated.html) (this is actually only available for Java 9+), but adding a dependency to "javax.annotation:jsr250-api" should also work (we currently get this as a transitive dependency for pmd-apex)

```
<groupId>javax.annotation</groupId>
<artifactId>jsr250-api</artifactId>
<version>1.0</version>
```

This has an annotation `javax.annotation.Generated` (without package "processing").

We might need to adjust our javacc-wrapper.xml / antlr-wrapper.xml ant scripts, to insert this annotation...

Reply:
Hm... it seems, that `@Generated` has only source retention and wouldn't be in the class file, which jacoco uses - so jacoco doesn't see this annotation...